tests/commonTemplates_test.go

```diff
@@ -19,9 +19,18 @@ import (
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"kubevirt.io/ssp-operator/internal/common"
 	commonTemplates "kubevirt.io/ssp-operator/internal/operands/common-templates"
 )
 
+var commonTemplateExpectedLabels = map[string]string{
+	common.AppKubernetesNameLabel:      "common-templates",
+	common.AppKubernetesManagedByLabel: "ssp-operator",
+	common.AppKubernetesPartOfLabel:    "test",
+	common.AppKubernetesVersionLabel:   "v0.0.0-test",
```

Comment:
The labels `common.AppKubernetesPartOfLabel` and `common.AppKubernetesVersionLabel` depend on the SSP resource. Could you change this variable into a function where these 2 labels are copied from the SSP resource? Or maybe, can you add these values to `TestSuiteStrategy` interface, then the function can be written like:

```
func commonTemplateExpectedLabels() map[string]string {
	return map[string]string{
		common.AppKubernetesNameLabel:      "common-templates",
		common.AppKubernetesManagedByLabel: "ssp-operator",
		common.AppKubernetesPartOfLabel:    strategy.getPartOfLabel(),
		common.AppKubernetesVersionLabel:   strategy.getVersionLabel(),
		common.AppKubernetesComponentLabel: common.AppComponentTemplating.String(),
	}
}
```

Reply:
I just changed that and almost fell into a trap.
When doing that, I have to explicitly set the labels on the SSP CR defined in the strategies otherwise this will not actually test anything (as the functions return "").
This means I wouldn't have to patch the SSP resource, but also means we're not testing the labels get set when SSP changes.

Reply:
Well yes, the default test strategy creates a new SSP CR and runs tests on it. So the labels need to be added to this CR. When running the tests on existing SSP CR, I would add code to `func (s *existingSspStrategy) Init()` that checks if the labels are present, and would fail the whole test suite if they are not.

Reply:
Cool, thanks. Will do

Reply:
Have a look now, I moved this into a function and used it everywhere. Also the labels are now set and verified as part of the strategy.
I think we can do without the assertions for every label, but for now they are there.